File: authentication/authentication/src/main/java/com/example/authentication/model/Payment.java
package com.example.authentication.model;

import com.example.authentication.entity.UserEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Payment {
    private Long paymentId;
    private Set<UserEntity> payedUsers;
    private String paymentDescription;
    private String paymentMethod; // Store payment methods: Cash, Credit Card, etc
    private String imageUrl;
    private LocalDateTime createAt;
    private LocalDateTime updateAt;
}

File: socket-server/model/Payment.cpp
#include "Payment.h"

Payment::Payment() : mobile(Mobile()),
                    paymentMethod("Cash"), paymentDescription("Cash method") {}

Payment::Payment(string paymentMethod, string paymentDescription) : mobile(Mobile()),
                    paymentMethod(paymentMethod), paymentDescription(paymentDescription) {}

string Payment::getPaymentMethod() {
    return paymentMethod;
}

string Payment::getPaymentDescription() {
    return paymentDescription;
}

void Payment::setPaymentMethod(string paymentMethod) {
    this->paymentMethod = paymentMethod;
}

void Payment::setPaymentDescription(string paymentDescription) {
    this->paymentDescription = paymentDescription;
}